src/MemStore.cc

```diff
@@ -323,7 +323,7 @@ MemStore::get(const cache_key *key)
     // XXX: We do not know the URLs yet, only the key, but we need to parse and
     // store the response for the Root().get() callers to be happy because they
     // expect IN_MEMORY entries to already have the response headers and body.
-    e->makeMemObject();
+    e->ensureMemObject(nullptr, nullptr, HttpRequestMethod());
```

Comment:
Something does not add up here. This `ensureMemObject(nullptr, nullptr...)` call will always trigger the (moved) `Bug: Missing MemObject::storeId` warning, right? I am sure you can adjust the code further to avoid that warning in this specific case, but does not this problem tell us that we cannot make the URIs/method trio constant?! AFAICT, here we are creating a StoreEntry with MemObject that lacks any URIs/method information. After this PR, this StoreEntry will remain in this half-baked state because nobody will be able to reset URIs/method. Would not exposing Store clients to such half-baked StoreEntry/MemObject objects create all sorts of problems?

I am guessing that before this PR the half-baked StoreEntry/MemObject  were updated with URIs/method information as soon as that info was loaded from the memory cache by/in MemStore::copyFromShm() or perhaps shortly after that.

If my suspicions are correct, then either we have to go back to mutable URIs/method trio (at least for the specific case where the original trio was empty) or we need to reshape this code so that it does not need the MemObject object until it knows what the actual URIs/method trio is. The latter approach would address the above XXX comment, so it is better in general, but I suspect it may not be feasible without revamping a lot of code. In that case, we would have to allow trio updates, at least from the empty to non-empty state.

There are other small problems with the recent changes, but let's address this big issue first.